src/Components/Store/Store.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useReducer } from 'react';
+import { Todo } from '../../types/Todo';
+import { getTodos } from '../../api/todos';
+
+export enum Actions {
+  addNew = 'addNew',
+  mark = 'mark',
+  destroy = 'destroy',
+  edit = 'edit',
+  destroyCompleted = 'destroyCompleted',
+  setNewUserId = 'setNewUserId',
+  setUserTodos = 'setUserTodos',
+  setLoadingError = 'setLoadingError',
+  setDeletingError = 'setDeletingError',
+  setUpdatingError = 'setUpdatingError',
+}
+
+export enum Keys {
+  Escape = 'Escape',
+  Enter = 'Enter',
+}
+
+type Action = { type: Actions.addNew, todo: Todo }
+| { type: Actions.mark, todo: Todo }
+| { type: Actions.destroy, todo: Todo }
+| { type: Actions.edit, todo: Todo }
+| { type: Actions.destroyCompleted }
+| { type: Actions.setNewUserId, userId: number }
+| { type: Actions.setUserTodos, todos: Todo[] }
+| { type: Actions.setLoadingError }
+| { type: Actions.setDeletingError }
+| { type: Actions.setUpdatingError };
+
+interface State {
+  userId: number,
+  allTodos: Todo[],
+  error: string,
+}
+
+function saveTodos(state: State, updatedTodos: Todo[]) {
+  return {
+    ...state,
+    allTodos: updatedTodos,
+  };
+}
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case Actions.mark: {
+      const updatedTodos = state.allTodos.map((todo: Todo) => {
+        return (todo.id === action.todo.id ? {
+          ...todo,
+          completed: !action.todo.completed,
+        } : todo);

```

Comment:
Try to rewrite like this
```suggestion
        return (
          todo.id === action.todo.id
            ? ({
                ...todo,
                completed: !action.todo.completed,
              })
            : todo
        );

```